projects/i18n/switch/language.switcher.ts

```diff
@@ -0,0 +1,46 @@
+import {Inject, inject, Injectable} from '@angular/core';
+import {LOCAL_STORAGE} from '@ng-web-apis/common';
+import {tuiIsPresent, tuiNormalizeCommonJSImport} from '@taiga-ui/cdk';
+import {TuiLanguage, TuiLanguageName} from '@taiga-ui/i18n/interfaces';
+import {TUI_DEFAULT_LANGUAGE} from '@taiga-ui/i18n/tools';
+import {BehaviorSubject, from, Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+import {TUI_LANG_DEFAULT_KEY, TUI_LANG_STORAGE_KEY} from './options';
+
+// @dynamic
+@Injectable({providedIn: `root`})
+export class TuiLanguageSwitcher extends BehaviorSubject<Observable<TuiLanguage>> {
```

Comment:
You extends `BehaviorSubject` but dont implement its behaviour.

It is good idea to use `BehaviorSubject` but let's do it in the same way as here (no public method except `next`):
https://github.com/Tinkoff/taiga-ui/blob/83a562e0bff72ad72f601a199e008d9696a7d287/projects/demo/src/modules/app/theme.service.ts#L7-L19